### ISSUE TITLE
Skill confirmation disappeared from more option

### ISSUE THREAD

**Issue by Andybui1012** [Fate-Grand-Automata/FGA#1705]
### Preparation

- [X] I tested the [latest release](https://github.com/Fate-Grand-Automata/FGA/releases/tag/canary)
- [X] I looked at other issues (even the [closed ones](https://github.com/Fate-Grand-Automata/FGA/issues?q=is%3Aissue+is%3Aclosed))
- [X] I read the [Troubleshooting Guide](https://github.com/Fate-Grand-Automata/FGA/wiki/Troubleshooting)

### FGO server

NA

### FGA build number

2265

### Describe the bug

I noticed that the app was clicking on the wrong servant and after checking the issue, i realized it is because of the skill confirmation being enable in FGA, but when i looked in more option, i couldn't find it.

### Video

![image](https://github.com/Fate-Grand-Automata/FGA/assets/31494945/aa08fd71-3652-4318-80b0-fb500464cc32)

### Device model

Nox

### Android version

7

**Comment by reconman** [comment]
FGA now auto-detects Skill Confirmation because too many messed up the setting. Had about 1 user report per day about an issue related to Skill Confirmation, no matter what I changed regarding the setting.

Did you enable Root for Screenshots according to https://github.com/Fate-Grand-Automata/FGA/wiki/Running-on-Emulators ?

If yes and it still doesn't work, post a video with debug mode.

**Comment by DAOWAce** [comment]
Not sure if this is what the OP means:

https://github.com/Fate-Grand-Automata/FGA/assets/480587/b50a7771-3548-4454-af4b-193dbdeb9681

I commented in another issue https://github.com/Fate-Grand-Automata/FGA/issues/1328 about something related to it, as they both seemed to happen around the same time.

Also separate issue, the script cannot enter battles that popup that "you'll set your squad before the battle" prompt; and there's quite a number of them in every part of the game.  Makes the automation part of things, well, quite a problem.

With this Christmas event, there's that pre-battle prompt, and forced usage of a 1 skill servant which seems to completely break FGA almost every single time (if set to spam all; doesn't see there's no skill there, can get stuck like in the video)..

**Comment by PKX** [comment]
Can confirm skills are now completely broken after this change: it always gets stuck at the same spot:
https://imgur.com/zKltn8z

With an old version, it works
https://imgur.com/QAYltGQ

**Comment by DAOWAce** [comment]
@ArthurKun21 Yeah, I thought Sishtar had to be manually added back when she got introduced into NA.  Not sure if that was the discussion, but remember reading her mentioned in these issues back on release.  If it was, then Emiya's update needs the same treatment.

But still, that wasn't really the point of the video, it's just the only one I had with the debug info on.  As mentioned, saber Okita (not alter) last event was triggering this extremely frequently, usually in wave 3.  I can see if I can find her for rental and cap another video for visuals as to exactly what's going on, but this is an issue other people have reported for a number of years now.  A few of them, not all the same cause though:

https://github.com/Fate-Grand-Automata/FGA/issues/248
https://github.com/Fate-Grand-Automata/FGA/issues/468
https://github.com/Fate-Grand-Automata/FGA/issues/1041
https://github.com/Fate-Grand-Automata/FGA/issues/1045
https://github.com/Fate-Grand-Automata/FGA/issues/1295
https://github.com/Fate-Grand-Automata/FGA/issues/1349

I have personally seen a few other issues over the years (not using skills, not using NP, getting stuck on other waves/servant selection, issues with story skip), but they seem to be resolved at this point.  The main problem at present really is this obtuse 'getting stuck on servant details'.  And the double-tapping thing.

It's also very frequently popping up the text "can't determine blah blah" when selecting cards, but still seems to select the right order.  I don't know why; I'd assume all the images are already there from JP which is 2 years ahead of NA.  Maybe some text is interfering?

Can confirm that old versions (like really old) did seem to work better in some areas, but they lacked the skill spam function _(and probably don't work anymore anyway)_, and I can't manually setup things for teams as I'm just not that kind of player.  Skill spam has been a **godsend**.  I can understand maybe setting up specific automation for farming specific stages without relying on rentals, but for use over the entire game? Just not feasible at all.

@reconman Does FGA have functionality to see skills are on cooldown now?  Was reading in older issues it didn't do that, but your comment if I'm understanding it right implies it does detect it now?  I thought last time I stopped/started script in battle it tried to use the skills all over again, so I'm not sure.

Anyway, give me a day or two and I'll try to get a new video recorded about the issue.

**Comment by DAOWAce** [comment]
Got it, and now I think I see what's happening.

https://github.com/Fate-Grand-Automata/FGA/assets/480587/3a583eea-625e-4bee-a305-6499e8eaee23

The detection box is a bit to the left of the skill number, and/or it looks like the OCR is tripping up on that specific icon.

This can be adjusted with the "game area mode", right?  I'm not sure how to use the options though.

**Comment by reconman** [comment]
> The only valid case for errors related to Skill Confirmation I saw was when you start the battle, stop the script and then start it again.

And skill spam only compares before/after images, it doesn't check for numbers right now.

FGA always assumes skills can be used when starting the script in battle.

**Comment by nguyentrong101094** [comment]
is there a previous release that work? Checked the releases page and the last release was 3 years ago.

**Comment by DAOWAce** [comment]
> > The only valid case for errors related to Skill Confirmation 

So I should open a new issue with a different title and copy/paste all this information over there, got it.
